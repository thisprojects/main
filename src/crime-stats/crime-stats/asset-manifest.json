{
  "files": {
    "main.css": "/crime-stats/static/css/main.5ce02d1d.chunk.css",
    "main.js": "/crime-stats/static/js/main.47483de9.chunk.js",
    "main.js.map": "/crime-stats/static/js/main.47483de9.chunk.js.map",
    "runtime-main.js": "/crime-stats/static/js/runtime-main.de4c9ff0.js",
    "runtime-main.js.map": "/crime-stats/static/js/runtime-main.de4c9ff0.js.map",
    "static/js/2.5126f43c.chunk.js": "/crime-stats/static/js/2.5126f43c.chunk.js",
    "static/js/2.5126f43c.chunk.js.map": "/crime-stats/static/js/2.5126f43c.chunk.js.map",
    "index.html": "/crime-stats/index.html",
    "precache-manifest.b214906e36a7e558e20d4cceca73f2a3.js": "/crime-stats/precache-manifest.b214906e36a7e558e20d4cceca73f2a3.js",
    "service-worker.js": "/crime-stats/service-worker.js",
    "static/css/main.5ce02d1d.chunk.css.map": "/crime-stats/static/css/main.5ce02d1d.chunk.css.map",
    "static/js/2.5126f43c.chunk.js.LICENSE": "/crime-stats/static/js/2.5126f43c.chunk.js.LICENSE"
  },
  "entrypoints": [
    "static/js/runtime-main.de4c9ff0.js",
    "static/js/2.5126f43c.chunk.js",
    "static/css/main.5ce02d1d.chunk.css",
    "static/js/main.47483de9.chunk.js"
  ]
}