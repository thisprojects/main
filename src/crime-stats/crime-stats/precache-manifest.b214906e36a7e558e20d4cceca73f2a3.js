self.__precacheManifest = (self.__precacheManifest || []).concat([
  {
    "revision": "36276d903ae060e58767a7c72813c90a",
    "url": "/crime-stats/index.html"
  },
  {
    "revision": "52090c35efbead8084d1",
    "url": "/crime-stats/static/css/main.5ce02d1d.chunk.css"
  },
  {
    "revision": "b4cd8cea65ca8463f428",
    "url": "/crime-stats/static/js/2.5126f43c.chunk.js"
  },
  {
    "revision": "4386a358a4fc57d2df32d9d927f6e8a1",
    "url": "/crime-stats/static/js/2.5126f43c.chunk.js.LICENSE"
  },
  {
    "revision": "52090c35efbead8084d1",
    "url": "/crime-stats/static/js/main.47483de9.chunk.js"
  },
  {
    "revision": "8a27c575ef54dcfafec9",
    "url": "/crime-stats/static/js/runtime-main.de4c9ff0.js"
  }
]);