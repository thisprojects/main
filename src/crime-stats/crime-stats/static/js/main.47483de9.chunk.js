(this["webpackJsonpcrime-stats"]=this["webpackJsonpcrime-stats"]||[]).push([[0],{15:function(e,t){e.exports="data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAADIAAAAyCAYAAAAeP4ixAAAAAXNSR0IArs4c6QAAAAlwSFlzAAALEwAACxMBAJqcGAAAAVlpVFh0WE1MOmNvbS5hZG9iZS54bXAAAAAAADx4OnhtcG1ldGEgeG1sbnM6eD0iYWRvYmU6bnM6bWV0YS8iIHg6eG1wdGs9IlhNUCBDb3JlIDUuNC4wIj4KICAgPHJkZjpSREYgeG1sbnM6cmRmPSJodHRwOi8vd3d3LnczLm9yZy8xOTk5LzAyLzIyLXJkZi1zeW50YXgtbnMjIj4KICAgICAgPHJkZjpEZXNjcmlwdGlvbiByZGY6YWJvdXQ9IiIKICAgICAgICAgICAgeG1sbnM6dGlmZj0iaHR0cDovL25zLmFkb2JlLmNvbS90aWZmLzEuMC8iPgogICAgICAgICA8dGlmZjpPcmllbnRhdGlvbj4xPC90aWZmOk9yaWVudGF0aW9uPgogICAgICA8L3JkZjpEZXNjcmlwdGlvbj4KICAgPC9yZGY6UkRGPgo8L3g6eG1wbWV0YT4KTMInWQAABbhJREFUaAXtmFuolUUUx+3mJdPEIC0KlaIsISjoQokvhkR2eSkpKIpAk4IoiCiIQgyEohsVPRToQ2DlSxGF0UsSCBLd7F5iUvRQUNnNsvvvd/b8c/z49tbO2VuPcRb8vzWzZs2adZmZ79t73Lj/CR0ygjgOZe5B4O8WG73GWtRHh8hgulGvsW5zhiUfSUWWsuJx4GNQO2z7BnAU2AIOBm1VQ7xvScfiaAJfjEznxBwgje+wcVfClf8CZhSZwUi1rY6kT88s0M1csqljte4P1YSFpZ3xC0v/K7jBSM43CXKRxNAcPGWxCdVSHmJJ2VagU08ByUCsyidA+TNAMoBUsrYV+0NKg3oku3NZ4AuwEUwti00sfC1chz0Hk4rsjCJTvrzIMnYmfav0CkhAAw8mmV/CojolNoMpILSMhvI/wYIivL3IfofPKzLZWWAHiC0vCSkJ6/QG9Ey2nsR+HHiX9pFlvZMr+WVF9miRvQ+PkwbhWYmN62hL2W6d3oCeBpFAXGI1iCNv0p6mELoaWIVUcCbte8G5QDKIH0HmemVLCbJeozPSx2eMu5iHN5mrg7m5Zb3Mq4deopMg3IqSZ+wwEP0E5VjfaXIXiw8h3wBmlXEdSjUU6ZwyIS0C74BrQRvlIkhQbTr/WZbM3MPMb8CL4DZwNsiCNP+lPS3eNj6d2Qa3CmwE3mKXAimV7/RG8Ewg72Ej2yH8c2T3g/FAvejS7Emp1vFoPQe+BrEZ/nCxEN3SHT6Lc6dg4nHwFtgJsqD8NCBFt9PrPK1Aswpx7ibGajvb6b8GVoBpQGrO7Uj38tmcbN8FQ74QzwMXAT8OvaHaqJ7n+fgDaCfyGbQfAD+D9WAT+BKEopd+X7jZdgslm3tr9NiGovO1pZNt5Jg63cbb5nSVdTtgvq3/KrO8Lv28kH8LdKCumjbs+7G4DuwA2wpXHoflAc0h+1nD/kDoBKxeA9aAj4AO+cXroZV0KJSMuqVymL2NHgGnR6nwVKkh7m83zt2HWR1v4ntks4EU3U5v1za8G0Fz3qvIrgL1u8kq9m1bYWs3inNvI40z22g/C24Ec0A3ylwr6Qej838Dbp3Y8nCbpHmgpr5XKc6cxCq3gvng8HpF2n4oLqpkbqtsLbMs+b7Q+QTkDZZ2gnoZmR+aE0Cor1WKUzEun1Q658A9JzpzRZEleHkCWVx0cgXHeS8Qq5S+/DOwEpwIaupLlXTKg6tjcsmv2PpTXGelySAvtDqoXBA6Xztu2+3WrJIVex74voodmkM+tCXXsb2mZLgZxPXFgkG8DraDU4ssW+UO+jrdrEAzqLYq5cV7TLEpG3YwyYqfK9lOOrFMq4UugMexy4vMF6nkNZ0K1gc++k2ujlX5FdRja+jnh9ywgnGPSjoYw0uHJLvOzKoy9h08b3UTkLlPl/HmQY+9cINoVu5DZCuA76HsDJojo+VMv7iY0GiqtZG2zmwoY5Fn4YVlvHnoE4C8rpZ6L4BLQJJBs/9UBzEL874gdWZlWapt8c1Fpy2YXARb0bkLzC12wrSX5EQ2Iq7BZDnOLkGWzJp5KTq2o3cLbfWaW6euxBtOKORloZ1hnYcY2RueDM1GeQtYD+J0vXjaMxnPZRHnw39iLMn4gPZUIMVepzfAZ5xMUC4VWb1sHFqDUIdz6LPNPHcPljHH3YZHAClzO70BPvcUhEtnq82nraOeiZyLT2mHVtNwfL8F01aFOBceHV+cOpr3xJ1FIZnf78HE4W48jnqFG4jbaieYDaSM2x7VwaQi03HUH1sGsw5I2Xr1Nh3VwSTrj+G8gZxvFFACsd0rmCkqQLVOR7KPn3F4Aetu6bF27WhdGf+WOrrMS4V7mBn8kD/SvMGkbg51C+aJzrTdqlhE+491CyIe1cGsRbgJ5JfqnubGxsB57WSvxaKn4xOLYmS95o3KsTr7B2wQyazB1AFFPsbHMjCWgbEMHOAZ+Af7xHmzL8KWcQAAAABJRU5ErkJggg=="},19:function(e,t,a){e.exports=a(44)},24:function(e,t,a){},43:function(e,t,a){},44:function(e,t,a){"use strict";a.r(t);var n=a(0),r=a.n(n),l=a(4),s=a.n(l),o=(a(24),a(5)),c=a.n(o),i=a(12),u=a(13),m=a(17),d=a(14),p=a(18),g=a(8),h=a(2),f=a(7);function E(e){return r.a.createElement("div",{id:"loading",className:e.loading?"show-loading":"hide-loading"},r.a.createElement("h1",null,"LOADING"),r.a.createElement(f.a,{icon:h.c,size:"3x",spin:!0}),r.a.createElement("p",null,"Major cities may take some time."))}function A(e){return r.a.createElement("div",{style:e.errors?{}:{display:"none"}},r.a.createElement("h3",null,e.errors))}function v(e){return e.filterState?r.a.createElement("div",{className:"filters"},r.a.createElement(y,{categories:Object.keys(e.filterState),filterState:e.filterState,totalCrimeCount:e.totalCrimeCount})):r.a.createElement("div",null,r.a.createElement("p",null,"Nothing To Filter"))}function y(e){return e.categories.map((function(t,a){return r.a.createElement("span",{key:a,className:"filter-container"},r.a.createElement("label",{key:a+1},r.a.createElement("input",{key:a+2,type:"checkbox",name:t,value:t}),r.a.createElement("span",{key:a+3,className:"filter-item-wrapper"},r.a.createElement("p",{key:a+4,className:"category-header"},t),r.a.createElement("p",{key:a+5,className:"category-count"},e.filterState[t].count," Counts"," "),r.a.createElement("p",{key:a+6,className:"category-percentage"},"(",(e.filterState[t].count/e.totalCrimeCount*100).toFixed(2)," ","%)"))))}))}var C=a(16),S="AIzaSyCyNv5BOZZfdKO3VDhQCOA3Ufm8tv8rCF8",b=function(){for(var e=document.querySelectorAll(".filter-container input"),t=0;t<e.length;t++)e[t].checked=!1},w=function(){var e=document.querySelector("#post-code").value;return fetch("https://maps.googleapis.com/maps/api/geocode/json?address=".concat(e,",uk&key=").concat(S)).then((function(e){return e.json()})).then((function(e){return e.results[0].geometry.location})).catch((function(e){return console.log("Location API Error - Invalid Location -----\x3e",e),"Location Error - Location is invalid. Remember only UK locations will return results."}))},k=function(e,t){var a=document.querySelector("#year-select").value,n=document.querySelector("#month-select").value;return fetch("https://data.police.uk/api/crimes-street/all-crime?lat=".concat(e,"&lng=").concat(t,"&date=").concat(a,"-").concat(n)).then((function(e){return e.json()})).catch((function(e){return console.log("Police API Error - Caught Error Is -----\x3e",e),"Crime API Error - Try an earlier month."}))},N=function(e,t){return e.location.street.name<t.location.street.name?-1:e.location.street.name>t.location.street.name?1:0},L=function(e,t){return null===e.outcome_status?1:null===t.outcome_status?-1:e.outcome_status.category<t.outcome_status.category?-1:e.outcome_status.category>t.outcome_status.category?1:0},I=function(e,t){var a=Object.keys(e),n=[];return a.forEach((function(a){!0===e[a].state&&n.push.apply(n,Object(C.a)(t.filter((function(e){return e.category===a}))))})),n.length>0?n:t},j=function(){for(var e=(new Date).getFullYear(),t=[e],a=0;a<10;a++)t.push(e-(a+1));return t};function Z(e){return r.a.createElement("div",{onClick:function(){return e.showAndHideSliders(e.selector,e.slideClass)},className:e.theClass},r.a.createElement(f.a,{icon:e.icon}),r.a.createElement("p",null,e.label))}function G(e){return r.a.createElement("div",{className:"postcode-picker"},r.a.createElement("input",{id:"post-code",type:"text",placeholder:"Location"}))}function R(e){return r.a.createElement("div",{className:"submit-button"},r.a.createElement("button",{onClick:function(){e.buttonClick()}},"Submit"))}var H=function(){return r.a.createElement("div",{className:"date-year"},r.a.createElement("span",null,"Year"),r.a.createElement("select",{id:"year-select",defaultValue:"2019"},r.a.createElement(O,{years:j()})))},O=function(e){return e.years.map((function(e){return r.a.createElement("option",{key:e,value:e},e)}))},W=function(){return r.a.createElement("div",{className:"date-month"},r.a.createElement("span",null,"Month"),r.a.createElement("select",{id:"month-select"},r.a.createElement(F,null)))},F=function(){return["01","02","03","04","05","06","07","08","09","10","11","12"].map((function(e){return r.a.createElement("option",{key:e,value:e},e)}))};function M(e){return e.results.length>0?e.results.map((function(e,t){return r.a.createElement("div",{key:t,className:"stat-row"},r.a.createElement("span",{className:"result-category"},e.category),r.a.createElement("span",{className:"result-location"},e.location.street.name),r.a.createElement("span",{className:"result-outcome"},null!==e.outcome_status?e.outcome_status.category:"No Outcome Recorded"))})):r.a.createElement("div",{className:"no-results"},r.a.createElement("p",null,"No Results - Click start to choose a location. Results are only available for UK locations"))}var x=function(e){return r.a.createElement("div",{className:"total-crimes"},r.a.createElement("p",null,"Total Crimes ",e.totalCrimes),r.a.createElement("input",{type:"radio",name:"sort",value:"location",onChange:function(){return e.locationSort()}}),r.a.createElement("p",null,"Location Sort"),r.a.createElement("input",{type:"radio",name:"sort",value:"outcome",onChange:function(){return e.outcomeSort()}}),r.a.createElement("p",null,"Outcome Sort"))};var U=function(e){return r.a.createElement("div",{className:"crime-stats"},r.a.createElement(x,{totalCrimes:e.results.length,locationSort:e.applyLocationSort,outcomeSort:e.applyOutcomeSort}),r.a.createElement("div",{className:"results-table"},r.a.createElement(M,{results:e.results})))},z=a(15),P=a.n(z),V=function(e){function t(e){var a;return Object(i.a)(this,t),(a=Object(m.a)(this,Object(d.a)(t).call(this,e))).setResults=function(e,t,n){var r;a.setState({crimeResults:e,filteredResults:e,filterState:(r=e,r.reduce((function(e,t){return e[t.category]?e[t.category].count++:(e[t.category]={},e[t.category].state=!1,e[t.category].count=1),e}),{})),errorState:t||null,loadingState:n||null})},a.showAndHideSliders=function(e,t){document.querySelector(e).classList.toggle(t)},a.applyLocationSort=function(){a.setState({filteredResults:a.state.filteredResults.sort(N)})},a.applyOutcomeSort=function(){a.setState({filteredResults:a.state.filteredResults.sort(L)})},a.state={lat:1,lng:52,crimeResults:[],filteredResults:[],filterState:{},errorState:null,loadingState:null},a}return Object(p.a)(t,e),Object(u.a)(t,[{key:"setFilterListeners",value:function(){for(var e=this,t=document.querySelectorAll(".filter-container input"),a=0;a<t.length;a++)!0===t[a].checked&&!1===this.state.filterState[t[a].name.state]&&(this.state.filterState[t[a].name].state=!0,this.applyFilters()),t[a].addEventListener("click",(function(t){var a=t.target;e.state.filterState[a.name].state=a.checked,e.applyFilters()}))}},{key:"callCrimeApi",value:function(e){var t;return c.a.async((function(a){for(;;)switch(a.prev=a.next){case 0:return a.next=2,c.a.awrap(k(this.state.lat,this.state.lng));case 2:t=a.sent,Array.isArray(t)?this.setResults(t,null,!1):this.setResults([],t,!1),b(),this.setFilterListeners(),e||this.showAndHideSliders("#header-wrapper","side-hide");case 7:case"end":return a.stop()}}),null,this)}},{key:"applyFilters",value:function(){this.setState({filteredResults:I(this.state.filterState,this.state.crimeResults)})}},{key:"onMarkerDragEnd",value:function(e){this.setState({lat:e.latLng.lat(),lng:e.latLng.lng(),loadingState:!0}),this.callCrimeApi(!0)}},{key:"render",value:function(){var e=this;return r.a.createElement("div",{id:"page-wrapper"},r.a.createElement(E,{loading:this.state.loadingState}),r.a.createElement("div",{id:"header-wrapper",className:"side-slide side-hide"},r.a.createElement(Z,{showAndHideSliders:this.showAndHideSliders,selector:"#header-wrapper",theClass:"header-stub",label:"Start",slideClass:"side-hide",icon:h.b}),r.a.createElement("div",{className:"header-active"},r.a.createElement("div",{className:"start-menu-inputs"},r.a.createElement(G,null),r.a.createElement(W,null),r.a.createElement(H,null)),r.a.createElement(R,{buttonClick:function(){var t;return c.a.async((function(a){for(;;)switch(a.prev=a.next){case 0:return a.next=2,c.a.awrap(w());case 2:(t=a.sent).lat&&t.lng?(e.setState({lat:t.lat,lng:t.lng,loadingState:!0}),e.callCrimeApi()):e.setResults([],t);case 4:case"end":return a.stop()}}))}}))),r.a.createElement("div",{id:"map",className:"map"},r.a.createElement(g.Map,{initialCenter:{lat:52.489471,lng:-1.898575},center:{lat:this.state.lat,lng:this.state.lng},google:this.props.google,zoom:13},r.a.createElement(g.Marker,{icon:P.a,position:{lat:this.state.lat,lng:this.state.lng},draggable:!0,onDragend:function(t,a,n){return e.onMarkerDragEnd(n)},name:"Current location"}))),r.a.createElement("div",{className:"filter-wrapper top-slide top-hide"},r.a.createElement(v,{filterState:this.state.filterState,totalCrimeCount:this.state.crimeResults.length})),r.a.createElement(Z,{showAndHideSliders:this.showAndHideSliders,selector:".filter-wrapper",theClass:"filter-stub",label:"Filters",slideClass:"top-hide",icon:h.a}),r.a.createElement("div",{className:"errors-wrapper",style:this.state.errorState?{display:"flex"}:{display:"none"}},r.a.createElement(A,{errors:this.state.errorState})),r.a.createElement("div",{className:"crime-stats-wrapper"},r.a.createElement(U,{results:this.state.filteredResults,applyLocationSort:this.applyLocationSort,applyOutcomeSort:this.applyOutcomeSort})))}}]),t}(n.Component),J=Object(g.GoogleApiWrapper)({apiKey:S})(V);a(43);var B=function(){return r.a.createElement("div",{className:"wrapper"},r.a.createElement(J,null))};Boolean("localhost"===window.location.hostname||"[::1]"===window.location.hostname||window.location.hostname.match(/^127(?:\.(?:25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?)){3}$/));s.a.render(r.a.createElement(B,null),document.getElementById("root")),"serviceWorker"in navigator&&navigator.serviceWorker.ready.then((function(e){e.unregister()}))}},[[19,1,2]]]);
//# sourceMappingURL=main.47483de9.chunk.js.map