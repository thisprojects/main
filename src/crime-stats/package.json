{
  "name": "crime-stats",
  "version": "0.1.0",
  "homepage": "http://www.nathandownes.co.uk/crime-stats",
  "private": true,
  "dependencies": {
    "@babel/plugin-proposal-class-properties": "^7.8.3",
    "@babel/preset-env": "^7.8.3",
    "@babel/preset-react": "^7.8.3",
    "@fortawesome/fontawesome-svg-core": "^1.2.26",
    "@fortawesome/free-solid-svg-icons": "^5.12.0",
    "@fortawesome/react-fontawesome": "^0.1.8",
    "@testing-library/react": "^9.4.0",
    "babel-jest": "^24.9.0",
    "google-maps-react": "^2.0.2",
    "load-google-maps-api": "^2.0.1",
    "react": "^16.9.0",
    "react-dom": "^16.9.0",
    "react-fontawesome": "^1.7.1",
    "react-scripts": "^3.3.0",
    "regenerator-runtime": "^0.13.3"
  },
  "scripts": {
    "start": "react-scripts start",
    "build": "react-scripts build",
    "test": "jest",
    "testwatch": "test -- --watch",
    "eject": "react-scripts eject"
  },
  "jest": {
    "moduleNameMapper": {
      "\\.(jpg|jpeg|png|gif|eot|otf|webp|svg|ttf|woff|woff2|mp4|webm|wav|mp3|m4a|aac|oga)$": "<rootDir>/__mocks__/fileMock.js",
      "\\.(css|less)$": "<rootDir>/__mocks__/styleMock.js"
    }
  },
  "eslintConfig": {
    "extends": "react-app"
  },
  "browserslist": {
    "production": [
      ">0.2%",
      "not dead",
      "not op_mini all"
    ],
    "development": [
      "last 1 chrome version",
      "last 1 firefox version",
      "last 1 safari version"
    ]
  },
  "devDependencies": {
    "@babel/plugin-transform-runtime": "^7.8.3",
    "enzyme": "^3.11.0",
    "enzyme-adapter-react-16": "^1.15.2",
    "identity-obj-proxy": "^3.0.0",
    "jest": "^24.9.0",
    "react-test-renderer": "^16.12.0"
  }
}
