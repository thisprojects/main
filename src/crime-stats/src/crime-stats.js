import React, { Component } from "react";
import { Map, GoogleApiWrapper, Marker } from "google-maps-react";
import {
  Filters,
  Errors,
  Loading
} from "./main-components/filters-errors-loading-components";
import {
  PostCode,
  Year,
  Month,
  Slider,
  SubmitButton
} from "./main-components/start-menu-components";
import {
  postCodeApiCall,
  policeApiCall,
  makeFilterStates,
  resetFilters,
  filterResults,
  locationSort,
  outcomeSort,
  key
} from "./js-functions/js-functions";
import CrimeStats from "./main-components/crime-components.js";
import { faFilter } from "@fortawesome/free-solid-svg-icons";
import { faPlayCircle } from "@fortawesome/free-solid-svg-icons";
import icon from "./pointer.png";

export class CrimeApp extends Component {
  constructor(props) {
    super(props);
    this.state = {
      lat: 1,
      lng: 52,
      crimeResults: [],
      filteredResults: [],
      filterState: {},
      errorState: null,
      loadingState: null
    };
  }

  setResults = (results, errors, loading) => {
    this.setState({
      crimeResults: results,
      filteredResults: results,
      filterState: makeFilterStates(results),
      errorState: errors || null,
      loadingState: loading || null
    });
  };

  setFilterListeners() {
    let x = document.querySelectorAll(".filter-container input");
    for (let i = 0; i < x.length; i++) {
      // Keep filter checkbox value and filter states in sync
      if (
        x[i].checked === true &&
        this.state.filterState[x[i].name.state] === false
      ) {
        this.state.filterState[x[i].name].state = true;
        this.applyFilters();
      }

      // Listen for filter element clicks
      x[i].addEventListener("click", e => {
        let t = e.target;
        this.state.filterState[t.name].state = t.checked;
        this.applyFilters();
      });
    }
  }

  async callCrimeApi(locationSetByDrag) {
    let results = await policeApiCall(this.state.lat, this.state.lng);

    //If a results array is returned, display the results. If not, display errors.
    Array.isArray(results)
      ? this.setResults(results, null, false)
      : this.setResults([], results, false);
    resetFilters();
    this.setFilterListeners();
    if (!locationSetByDrag)
      this.showAndHideSliders("#header-wrapper", "side-hide");
  }

  applyFilters() {
    this.setState({
      filteredResults: filterResults(
        this.state.filterState,
        this.state.crimeResults
      )
    });
  }

  onMarkerDragEnd(evt) {
    this.setState({
      lat: evt.latLng.lat(),
      lng: evt.latLng.lng(),
      loadingState: true
    });
    this.callCrimeApi(true);
  }

  showAndHideSliders = (selector, slideClass) => {
    let x = document.querySelector(selector);
    x.classList.toggle(slideClass);
  };

  applyLocationSort = () => {
    this.setState({
      filteredResults: this.state.filteredResults.sort(locationSort)
    });
  };

  applyOutcomeSort = () => {
    this.setState({
      filteredResults: this.state.filteredResults.sort(outcomeSort)
    });
  };

  render() {
    const submitButton = async () => {
      let location = await postCodeApiCall();

      // If a valid location is returned, set lat and lng then call crime API. If location is invalid, display errors.
      if (location.lat && location.lng) {
        this.setState({
          lat: location.lat,
          lng: location.lng,
          loadingState: true
        });
        this.callCrimeApi();
      } else {
        this.setResults([], location);
      }
    };

    return (
      <div id="page-wrapper">
        <Loading loading={this.state.loadingState} />
        <div id="header-wrapper" className="side-slide side-hide">
          <Slider
            showAndHideSliders={this.showAndHideSliders}
            selector={"#header-wrapper"}
            theClass={"header-stub"}
            label={"Start"}
            slideClass={"side-hide"}
            icon={faPlayCircle}
          />
          <div className="header-active">
            <div className="start-menu-inputs">
              <PostCode />
              <Month />
              <Year />
            </div>
            <SubmitButton buttonClick={submitButton} />
          </div>
        </div>
        <div id="map" className="map">
          <Map
            initialCenter={{
              lat: 52.489471,
              lng: -1.898575
            }}
            center={{
              lat: this.state.lat,
              lng: this.state.lng
            }}
            google={this.props.google}
            zoom={13}
          >
            <Marker
              icon={icon}
              position={{ lat: this.state.lat, lng: this.state.lng }}
              draggable={true}
              onDragend={(map, t, coord) => this.onMarkerDragEnd(coord)}
              name={"Current location"}
            />
          </Map>
        </div>
        <div className="filter-wrapper top-slide top-hide">
          <Filters
            filterState={this.state.filterState}
            totalCrimeCount={this.state.crimeResults.length}
          />
        </div>
        <Slider
          showAndHideSliders={this.showAndHideSliders}
          selector={".filter-wrapper"}
          theClass={"filter-stub"}
          label={"Filters"}
          slideClass={"top-hide"}
          icon={faFilter}
        />
        <div
          className="errors-wrapper"
          style={
            this.state.errorState ? { display: "flex" } : { display: "none" }
          }
        >
          <Errors errors={this.state.errorState} />
        </div>
        <div className="crime-stats-wrapper">
          <CrimeStats
            results={this.state.filteredResults}
            applyLocationSort={this.applyLocationSort}
            applyOutcomeSort={this.applyOutcomeSort}
          />
        </div>
      </div>
    );
  }
}
export default GoogleApiWrapper({
  apiKey: key
})(CrimeApp);
