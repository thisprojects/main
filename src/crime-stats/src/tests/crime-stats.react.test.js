import React from 'react';
import { CrimeApp } from '../crime-stats.js';
import renderer from 'react-test-renderer';

test('Main Component Matches Snapshot', () => {
  const component = renderer.create(
    <CrimeApp />,
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});



