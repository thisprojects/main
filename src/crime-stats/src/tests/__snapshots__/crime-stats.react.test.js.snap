// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Main Component Matches Snapshot 1`] = `
<div
  id="page-wrapper"
>
  <div
    className="hide-loading"
    id="loading"
  >
    <h1>
      LOADING
    </h1>
    <svg
      aria-hidden="true"
      className="svg-inline--fa fa-spinner fa-w-16 fa-spin fa-3x "
      data-icon="spinner"
      data-prefix="fas"
      focusable="false"
      role="img"
      style={Object {}}
      viewBox="0 0 512 512"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M304 48c0 26.51-21.49 48-48 48s-48-21.49-48-48 21.49-48 48-48 48 21.49 48 48zm-48 368c-26.51 0-48 21.49-48 48s21.49 48 48 48 48-21.49 48-48-21.49-48-48-48zm208-208c-26.51 0-48 21.49-48 48s21.49 48 48 48 48-21.49 48-48-21.49-48-48-48zM96 256c0-26.51-21.49-48-48-48S0 229.49 0 256s21.49 48 48 48 48-21.49 48-48zm12.922 99.078c-26.51 0-48 21.49-48 48s21.49 48 48 48 48-21.49 48-48c0-26.509-21.491-48-48-48zm294.156 0c-26.51 0-48 21.49-48 48s21.49 48 48 48 48-21.49 48-48c0-26.509-21.49-48-48-48zM108.922 60.922c-26.51 0-48 21.49-48 48s21.49 48 48 48 48-21.49 48-48-21.491-48-48-48z"
        fill="currentColor"
        style={Object {}}
      />
    </svg>
    <p>
      Major cities may take some time.
    </p>
  </div>
  <div
    className="side-slide side-hide"
    id="header-wrapper"
  >
    <div
      className="header-stub"
      onClick={[Function]}
    >
      <svg
        aria-hidden="true"
        className="svg-inline--fa fa-play-circle fa-w-16 "
        data-icon="play-circle"
        data-prefix="fas"
        focusable="false"
        role="img"
        style={Object {}}
        viewBox="0 0 512 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M256 8C119 8 8 119 8 256s111 248 248 248 248-111 248-248S393 8 256 8zm115.7 272l-176 101c-15.8 8.8-35.7-2.5-35.7-21V152c0-18.4 19.8-29.8 35.7-21l176 107c16.4 9.2 16.4 32.9 0 42z"
          fill="currentColor"
          style={Object {}}
        />
      </svg>
      <p>
        Start
      </p>
    </div>
    <div
      className="header-active"
    >
      <div
        className="start-menu-inputs"
      >
        <div
          className="postcode-picker"
        >
          <input
            id="post-code"
            placeholder="Location"
            type="text"
          />
        </div>
        <div
          className="date-month"
        >
          <span>
            Month
          </span>
          <select
            id="month-select"
          >
            <option
              value="01"
            >
              01
            </option>
            <option
              value="02"
            >
              02
            </option>
            <option
              value="03"
            >
              03
            </option>
            <option
              value="04"
            >
              04
            </option>
            <option
              value="05"
            >
              05
            </option>
            <option
              value="06"
            >
              06
            </option>
            <option
              value="07"
            >
              07
            </option>
            <option
              value="08"
            >
              08
            </option>
            <option
              value="09"
            >
              09
            </option>
            <option
              value="10"
            >
              10
            </option>
            <option
              value="11"
            >
              11
            </option>
            <option
              value="12"
            >
              12
            </option>
          </select>
        </div>
        <div
          className="date-year"
        >
          <span>
            Year
          </span>
          <select
            defaultValue="2019"
            id="year-select"
          >
            <option
              value={2020}
            >
              2020
            </option>
            <option
              value={2019}
            >
              2019
            </option>
            <option
              value={2018}
            >
              2018
            </option>
            <option
              value={2017}
            >
              2017
            </option>
            <option
              value={2016}
            >
              2016
            </option>
            <option
              value={2015}
            >
              2015
            </option>
            <option
              value={2014}
            >
              2014
            </option>
            <option
              value={2013}
            >
              2013
            </option>
            <option
              value={2012}
            >
              2012
            </option>
            <option
              value={2011}
            >
              2011
            </option>
            <option
              value={2010}
            >
              2010
            </option>
          </select>
        </div>
      </div>
      <div
        className="submit-button"
      >
        <button
          onClick={[Function]}
        >
          Submit
        </button>
      </div>
    </div>
  </div>
  <div
    className="map"
    id="map"
  >
    <div
      style={
        Object {
          "height": "100%",
          "position": "absolute",
          "width": "100%",
        }
      }
    >
      <div
        style={
          Object {
            "bottom": 0,
            "display": "inherit",
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          }
        }
      >
        Loading map...
      </div>
    </div>
  </div>
  <div
    className="filter-wrapper top-slide top-hide"
  >
    <div
      className="filters"
    />
  </div>
  <div
    className="filter-stub"
    onClick={[Function]}
  >
    <svg
      aria-hidden="true"
      className="svg-inline--fa fa-filter fa-w-16 "
      data-icon="filter"
      data-prefix="fas"
      focusable="false"
      role="img"
      style={Object {}}
      viewBox="0 0 512 512"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M487.976 0H24.028C2.71 0-8.047 25.866 7.058 40.971L192 225.941V432c0 7.831 3.821 15.17 10.237 19.662l80 55.98C298.02 518.69 320 507.493 320 487.98V225.941l184.947-184.97C520.021 25.896 509.338 0 487.976 0z"
        fill="currentColor"
        style={Object {}}
      />
    </svg>
    <p>
      Filters
    </p>
  </div>
  <div
    className="errors-wrapper"
    style={
      Object {
        "display": "none",
      }
    }
  >
    <div
      style={
        Object {
          "display": "none",
        }
      }
    >
      <h3 />
    </div>
  </div>
  <div
    className="crime-stats-wrapper"
  >
    <div
      className="crime-stats"
    >
      <div
        className="total-crimes"
      >
        <p>
          Total Crimes 
          0
        </p>
        <input
          name="sort"
          onChange={[Function]}
          type="radio"
          value="location"
        />
        <p>
          Location Sort
        </p>
        <input
          name="sort"
          onChange={[Function]}
          type="radio"
          value="outcome"
        />
        <p>
          Outcome Sort
        </p>
      </div>
      <div
        className="results-table"
      >
        <div
          className="no-results"
        >
          <p>
            No Results - Click start to choose a location. Results are only available for UK locations
          </p>
        </div>
      </div>
    </div>
  </div>
</div>
`;
